pkgs/by-name/ex/extract-vmlinux/package.nix

```diff
@@ -0,0 +1,58 @@
+# Extracts <linux-latest-src>/scripts/extract-vmlinux as package. As this script
+# is rather basic and generic, we do not need to care about the specific Linux
+# src version from that we get the script.
+
+{ lib
+, linux_latest # kernel package to get the source tree from
+, makeWrapper
+, runCommandLocal
+
+, binutils # readelf
+, coreutils
+, gnugrep
+
+  # decompressors for possible kernel image formats
+, bzip2
+, gzip
+, lz4
+, lzop
+, xz
+, zstd
+}:
+
+let
+  deps = [
+    binutils
+    coreutils
+    gnugrep
+    gzip
+    xz
+    bzip2
+    lzop
+    lz4
+    zstd
+  ];
+in
+runCommandLocal "extract-vmlinux-${toString linux_latest.version}"
+{
+  inherit (linux_latest) version;
+  nativeBuildInputs = [ makeWrapper ];
+  meta = with lib; {
+    homepage = "https://www.kernel.org/";
+    description = "Extract the vmlinux (ELF) from an compressed kernel image (bzImage)";
+    license = licenses.gpl2Only;
+    maintainers = [ maintainers.phip1611 ];
+    platforms = platforms.all;
+    mainProgram = "extract-vmlinux";
+  };
+} ''
+  set -euo pipefail
+
+  mkdir -p $out/bin
+
+  tar xf ${linux_latest.src} # untar the .tar.xz file
```

Comment:
unpackFile?  Or you could use `mkDerivation` and then just inherit the kernel source.